core/state_transition.go

```diff
@@ -245,127 +245,55 @@ func (st *StateTransition) to() common.Address {
 	return *st.msg.To
 }
 
-var SoulGasTokenABI abi.ABI
-
-func init() {
-	var err error
-	SoulGasTokenABI, err = util.ParseFunctionsAsABI([]string{
-		"function deposit()", // used by op_geth_test.go
-		"function balanceOf(address account) returns (uint256 balance)",
-		"function burnFrom(address account, uint256 value)",
-		"function batchMint(address[] accounts, uint256[] values)"})
-	if err != nil {
-		panic(err)
-	}
-}
-
-func getSoulBalanceData(account common.Address) []byte {
-	method, ok := SoulGasTokenABI.Methods["balanceOf"]
-	if !ok {
-		panic("balanceOf method not found")
-	}
-
-	argument, err := method.Inputs.Pack(account)
-	if err != nil {
-		panic("failed to pack argument")
-	}
-
-	data := make([]byte, len(method.ID)+len(argument))
-	copy(data[0:len(method.ID)], method.ID)
-	copy(data[len(method.ID):], argument)
-	return data
-}
-
-func parseSoulBalanceResp(ret []byte) (*uint256.Int, error) {
-	method, ok := SoulGasTokenABI.Methods["balanceOf"]
-	if !ok {
-		panic("balanceOf method not found")
-	}
+const (
+	// should keep it in sync with the balances field of SoulGasToken contract
+	balancesSlot = uint64(51)
```

Comment:
I am thinking whether we can get the slot by an external call as
```
 function getSlot() public pure returns (uint256) {
        uint256 slot;
        assembly {
            slot := _balances.slot
        }
        return slot;
    }
```

but definitively, it requires a lot of changes of code

Reply:
Gotcha, I think we can add a test in the monorepo that automatically checks for this.